src/app.py

```diff
@@ -45,24 +57,71 @@ def gather_articles():
         # Need to unwrap ObjectID objects from MongoDB into str ids
         article_ids = [str(article) for article in result.values()]
-        try:
-            logging.info(f"Sending notification for {len(article_ids)} articles")
-            requests.post(
-                ("http://" + VOLUME_NOTIFICATIONS_ENDPOINT), data={"articleIDs": article_ids}
-            )
-        except:
-            logging.error("Unable to connect to volume-backend.")
+        if article_ids:
+            try:
+                logging.info(f"Sending notification for {len(article_ids)} articles")
+                requests.post(
+                    VOLUME_NOTIFICATIONS_ENDPOINT + "/articles/",
+                    data={"articleIDs": article_ids},
+                )
+            except Exception as e:
+                logging.error("Articles unable to connect to volume-backend.")
+
+
+# Function for gathering magazines for running with scheduler
+def gather_magazines():
+    logging.info("Gathering magazines")
+    magazines = []
+    with MongoClient(MONGO_ADDRESS) as client:
+        db = client[DATABASE]
+        data = sheet.get_all_values()
+        len_data = len(data)
+        parse_counter = 1
+        for i in range(1, len_data):
+            if data[i][5] != "1" and data[i][0] != "":
+                p = list(
+                    filter(lambda p: p["slug"] == data[i][1], publications_serialized)
+                )
+                p = p[0] if p else None
+                magazines.append(Magazine(data[i], p).serialize())
+                sheet.update_cell(i + 1, 6, 1)
+            else:
+                parse_counter += 1
+        if parse_counter < len_data:
+            magazine_upserts = [
+                UpdateOne({"pdfURL": m["pdfURL"]}, {"$set": m}, upsert=True)
+                for m in magazines
+            ]
+            # Add magazines to db
+            result = db.magazines.bulk_write(
+                magazine_upserts, ordered=False
+            ).upserted_ids
+
+            # Need to unwrap ObjectID objects from MongoDB into str ids
+            magazine_ids = [str(magazine) for magazine in result.values()]
+            try:
+                logging.info(f"Sending notification for {len(magazine_ids)} magazines")
+
+                requests.post(
+                    VOLUME_NOTIFICATIONS_ENDPOINT + "/magazines/",
+                    data={"magazineIDs": magazine_ids},
+                )
+            except Exception as e:
+                logging.error("Magazines unable to connect to volume-backend.")
 
 
 # Before first run, clear states
 for f in os.listdir(STATES_LOCATION):
     os.remove(os.path.join(STATES_LOCATION, f))
 
 # Get initial refresh
+gather_magazines()
 gather_articles()
 
-# Schedule the function to run every 1 hour
+
+# Schedule the function to run every 10 minutes
 schedule.every(10).minutes.do(gather_articles)
+schedule.every(10).minutes.do(gather_magazines)
```

Comment:
This is up to you but maybe we want to just write a small function called like `gather_all_content` that calls both gather_magazines and gather_articles and then just schedule that one function? This way if we want to add more types of content ingestion in the future it's just a one line function call addition instead of another task scheduled

Reply:
gather_tiktoks()